<launch>
    <arg name="run_tag_detector" default="true"/>

    <!-- Nodelet Manager -->
    <node pkg="nodelet" type="nodelet" name="zed_nodelet_manager" args="manager" output="screen"/>

    <node if="$(arg run_tag_detector)"
          pkg="nodelet" type="nodelet" name="zed_nodelet"
          args="load zed_nodelets/ZEDWrapperNodelet zed_nodelet_manager" output="screen">
        <rosparam file="$(find mrover)/config/zed/base.yaml" command="load"/>
        <rosparam file="$(find mrover)/config/zed/pos_tracking.yaml" command="load"/>
        <rosparam file="$(find mrover)/config/zed/zed2i.yaml" command="load"/>

        <remap to="/camera/color/image_raw" from="/zed_nodelet/rgb/image_rect_color"/>
        <remap to="/camera/depth/points" from="/zed_nodelet/point_cloud/cloud_registered"/>
        <remap to="/camera/color/camera_info" from="/zed_nodelet/rgb/camera_info"/>
    </node>

    <!-- Tag detector as nodelet -->
    <node pkg="nodelet" type="nodelet" name="tag_detector"
          args="load mrover/TagDetectorNodelet zed_nodelet_manager" output="screen">
    </node>

    <!-- Static TF publisher for ZED mount-->
    <node pkg="tf" type="static_transform_publisher" name="zed_mount_link_publisher"
          args="0 0 0 0 0 0 base_link zed_mount_link 100"/>

    <!-- Load ZED URDF -->
    <param name="zed2i_description" command="$(find xacro)/xacro '$(find zed_wrapper)/urdf/zed_descr.urdf.xacro'
                        camera_name:=zed2i
                        camera_model:=zed2i
                        base_frame:=base_link"/>

    <!-- ZED URDF publisher -->
    <node name="zed2i_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"
          required="true">
        <remap from="robot_description" to="zed2i_description"/>
    </node>
</launch>
